File: progs/python/trio/exn.py
#Exception: reraise
import trio
from random import randint

counter = 0

async def write(msg):
    print(msg)

async def async_int():
    async def inner():
        raise Exception("argh")
    async with trio.open_nursery() as nursery:
        nursery.start_soon(inner)
    return 0

async def main():
    a = async_int()
    b = async_int()
    print("working")
    c = await a + await b
    await write(c)

trio.run(main)

File: progs/javascript/exn.js
//Exception: reraise
let counter = 0;

async function print(msg, signal) {
  console.log(msg);
}

async function asyncInt(signal) {
  let inner = async () => {
    throw new Error("argh");
  };
  inner();

  return 0;
}

async function main() {
  let a = asyncInt();
  let b = asyncInt();
  console.log("working");
  let c = await a + await b;
  await print(c);
}

main().then(() => console.log("exiting"))
